File: src/common/auth/context.go
package auth

import (
	"context"
	"errors"

	auth "github.com/cshep4/premier-predictor-microservices/src/common/auth/internal/context"

	"google.golang.org/grpc/metadata"
)

func MetadataFromContext(ctx context.Context) (context.Context, error) {
	token, ok := auth.GetTokenFromContext(ctx)
	if !ok {
		return nil, errors.New("missing token")
	}

	return metadata.AppendToOutgoingContext(context.Background(), "token", token), nil
}

File: src/gatewayservice/src/graphql/model/context/context.ts
import * as grpc from "grpc";
import WebSocket from "ws";

export interface TokenContext {
    token: string;
}

export interface SubscriptionContext {
    token: string;
    webSocket: WebSocket;
}

export function tokenMetadata(ctx: TokenContext | SubscriptionContext, audience: string = ""): grpc.Metadata {
    const md = new grpc.Metadata();
    md.add('token', ctx.token);
    md.add('audience', audience);

    return md
}